src/sync/index.js

```diff
@@ -83,259 +84,23 @@ async function sync(db){
-        console.error(`Error: ${err.message}`);
-        resolve();
-        return;
-      }
-
-      console.log(`${startBlock} - ${endBlock}: Retrieved ${result.length} entries from TopicCreated`);
-      var createTopicPromises = [];
-
-      _.forEach(result, (event, index) => {
-        let blockNum = event.blockNumber;
-        let txid = event.transactionHash;
-        _.forEachRight(event.log, (rawLog) => {
-          if(rawLog['_eventName'] === 'TopicCreated'){
-            var insertTopicDB = new Promise(async (resolve) =>{
-              try {
-                var topic = new Topic(blockNum, txid, rawLog).translate();
-                await db.Topics.insert(topic);
-                resolve();
-              } catch(err) {
-                console.error(`Error: ${err.message}`);
-                resolve();
-                return;
-              }
-            });
-
-            createTopicPromises.push(insertTopicDB);
-          }
-        });
-      });
-
-      Promise.all(createTopicPromises).then(()=>{
-        resolve()
-      });
+    var syncTopicCreatedPromise = new Promise(async (resolve) => { 
+      syncTopicCreated(resolve, db, startBlock, endBlock, removeHexPrefix); 
     });
-
-    // sync CentralizedOracleCreated
-    var syncCentralizedOracleCreatedPromise = new Promise(async (resolve) => {
-      let result;
-      try {
-        result = await qclient.searchLogs(startBlock, endBlock, Contracts.EventFactory.address, [Contracts.OracleFactory.CentralizedOracleCreated], Contracts, removeHexPrefix);
-        console.log('searchlog CentralizedOracleCreated')
-      } catch(err) {
-        console.error(`Error: ${err.message}`);
-        resolve();
-        return;
-      }
-
-      console.log(`${startBlock} - ${endBlock}: Retrieved ${result.length} entries from CentralizedOracleCreated`);
-      var createCentralizedOraclePromises = [];
-
-      _.forEach(result, (event, index) => {
-        let blockNum = event.blockNumber;
-        let txid = event.transactionHash;
-        _.forEachRight(event.log, (rawLog) => {
-          if(rawLog['_eventName'] === 'CentralizedOracleCreated'){
-            var insertOracleDB = new Promise(async (resolve) =>{
-              try {
-                var centralOracle = new CentralizedOracle(blockNum, txid, rawLog).translate();
-                await db.Oracles.insert(centralOracle);
-                resolve();
-              } catch(err) {
-                console.error(`Error: ${err.message}`);
-                resolve();
-                return;
-              }
-            });
-
-            createCentralizedOraclePromises.push(insertOracleDB);
-          }
-        });
-      });
-
-      Promise.all(createCentralizedOraclePromises).then(()=>{
-        resolve()
-      });
+    var syncCentralizedOracleCreatedPromise = new Promise(async (resolve) => { 
+      syncCentralizedOracleCreated(resolve, db, startBlock, endBlock, removeHexPrefix, oraclesNeedInfoUpdate);
```

Comment:
1. `async` without `await`?
2. `resolve` as last parameter

Reply:
seems fine without awaiting. i think the await is handled in the Promise.all => then()